packages/side-nav/src/vaadin-side-nav-item.js

```diff
@@ -183,24 +183,37 @@ class SideNavItem extends ElementMixin(ThemableMixin(PolylitMixin(LitElement)))
   /** @protected */
   render() {
     return html`
-      <a href="${ifDefined(this.path)}" part="item" aria-current="${this.active ? 'page' : 'false'}">
-        <slot name="prefix"></slot>
-        <slot></slot>
-        <slot name="suffix"></slot>
+      <div part="item" @click="${this._onItemClick}">
+        <a href="${ifDefined(this.path)}" part="link" aria-current="${this.active ? 'page' : 'false'}">
+          <slot name="prefix"></slot>
+          <slot></slot>
+          <slot name="suffix"></slot>
+        </a>
         <button
           part="toggle-button"
           @click="${this.__toggleExpanded}"
           aria-controls="children"
           aria-expanded="${this.expanded}"
           aria-label="Toggle child items"
         ></button>
-      </a>
+      </div>
       <ul part="children" ?hidden="${!this.expanded}">
         <slot name="children"></slot>
       </ul>
     `;
   }
 
+  /** @private */
+  _onItemClick(event) {
+    const item = event.currentTarget;
+    const [link, button] = item.children;
+
+    // Toggle item expanded state unless the click event is captured by the link
+    if (!link.hasAttribute('href') && this.hasAttribute('has-children')) {
+      button.click();
+    }
+  }
+
   /** @private */
   __toggleExpanded(e) {
     e.preventDefault();
```

Comment:
Calling `stopPropagation()` on the button click is now necessary to prevent the event from being handled twice (added a comment to make it clear). Renamed `e.preventDefault()` as it seems unnecessary to me.

Reply:
Right, thanks for clarifying.